src/core/mac/mac.cpp

```diff
@@ -1091,6 +1091,10 @@ void Mac::ReceiveDoneTask(Frame *aFrame, ThreadError aError)
         break;
     }
 
+    // Duplicate Address Protection
+    VerifyOrExit(memcmp(&srcaddr.mExtAddress, &mExtAddress, sizeof(srcaddr.mExtAddress)) != 0,
```

Comment:
well, I might have to be after the switch in case we got short address. Thus after that, we have whitelist, blacklist filtering and then duplicate check. So perhaps move it before the whitelist/blacklist?